week-3/implement/is-proper-fraction.js

```diff
@@ -5,31 +5,60 @@
 // Fractions: https://www.bbc.co.uk/bitesize/topics/zt9n6g8/articles/zjxpp4j
 // Written here like this: 1/2 == Numerator/Denominator
 
+function isProperFraction(numerator, denominator) {
+  const fraction = numerator / denominator;
+  if (denominator === 0) {
+    return "Error (Denominator cannot be zero)";
+  } else if (
+    (fraction < 1 && fraction > 0) ||
+    (fraction > -1 && fraction < 0)
+  ) {
+    return true;
+  } else {
+    return false;
+  }
+}
```

Comment:
It's all correct. But I have some ideas how to make it a little shorter:
```
  if (denominator === 0) {
    return "Error (Denominator cannot be zero)";
  }
  
  return (fraction > -1 && fraction < 1 && fraction !== 0);
}
```

Let me explain you a couple of things here.
First, If the first check is `true` (`denominator === 0`) then the function returns an error and the rest of it won't be performed. If it is `false` then the function continue to work anyway. It means that we don't really need `else` here.

Second point is that comparison like `fraction < 1` results in either `true` or `false`. It mean that we don't really need to write something like:
```
if (fraction > 1) {
  return true;
} else {
  return false;
}
```
We can just write `return fraction > 1` and that's it.

In this task we have few conditions to check, but the approach may be the same: `return (fraction > -1 && fraction < 1 && fraction !== 0);` because the result of these 3 checks will be either `true` or `false`.

Reply:
Using return to break the loop. That is brilliant 🤩 . I understand if the comparison returns true or false, there is no need to check with if statement.